compiler/vmgen.nim

```diff
@@ -466,9 +466,12 @@ proc sameConstant*(a, b: PNode): bool =
     of nkEmpty: result = true
     else:
       if a.len == b.len:
-        for i in 0..<a.len:
-          if not sameConstant(a[i], b[i]): return
-        result = true
+        if first and cyclicTree(a):
```

Comment:
not sure if this is correct, maybe instead we could do this differently:

in https://github.com/nim-lang/Nim/pull/15724/files#diff-43e428ef9f574e1c6c74da514b5f337286e4402916c094bac1c8e21ed6aca36bR998 instead of:

`elif sameConstant(nb, nc): ret = true`

do something that only considers procvar's

Reply:
I'm quite sure `cyclicTree` is buggy and this is a strange solution. Instead special case tyProc or similar.

Reply:
done